ttnn/cpp/ttnn/operations/reduction/argmax/argmax_composite.cpp

```diff
@@ -0,0 +1,141 @@
+    Tensor masked_input = tt::numpy::mask_padded_input<::bfloat16>(padded_shape, unpadded_shape, DataType::BFLOAT16);
+    masked_input = where(masked_input, input_a, t_inf, output_mem_config.value());
+    return masked_input;
+}
+// Argmax returns the index of maximum element in the tensor
+Tensor _argmax(const Tensor& input_t, int64_t _dim, bool all, const std::optional<MemoryConfig>& output_mem_config) {
+    auto output_memory_config = output_mem_config.value_or(input_t.memory_config());
+    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_t}))};
+    operation::launch_with_autoformat(
+        [_dim, all, output_memory_config](
+            const std::vector<Tensor>& input_tensors,
+            const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+            const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
+            const auto& input = input_tensors.at(0);
+            auto& input_shape = input.get_legacy_shape();
+            TT_FATAL(input_shape.rank() == 4, "supported for rank-4 tensors at this time");
+
+            Tensor input_a = create_mask(input, output_memory_config);
+
+            uint32_t dim = input_shape.get_normalized_index(_dim);
+            int size = input_a.volume();
+
+            if (!all) {
+                if ((dim == (input_shape.rank() - 1)) || (dim == (input_shape.rank() - 2))) {
+                    bool is_width = (dim == (input_shape.rank() - 1));
+                    Tensor max_val = max(input_a, dim, output_memory_config);
+                    Tensor max_tensor = ttnn::zeros_like(input_a);
+                    Tensor tindex = tt::numpy::index_width<::bfloat16>(
+                        input_shape, DataType::BFLOAT16, Layout::TILE, input_a.device(), output_memory_config);
+                    if (is_width) {
+                        max_tensor = ttnn::add(max_tensor, max_val, std::nullopt, output_memory_config);
+                    } else {
+                        tindex = tt::numpy::index_height<::bfloat16>(
+                            input_shape, DataType::BFLOAT16, Layout::TILE, input_a.device(), output_memory_config);
+                        max_tensor = ttnn::add(max_tensor, max_val, std::nullopt, output_memory_config);
+                    }
+                    tindex = tindex.to(input_a.device());
+                    max_val.deallocate();
+                    Tensor cmp_results = ttnn::eq(input_a, max_tensor, std::nullopt, output_memory_config);
+                    max_tensor.deallocate();
+                    Tensor max_indices = ttnn::multiply(cmp_results, tindex, std::nullopt, output_memory_config);
+                    cmp_results.deallocate();
+                    Tensor result = where(ttnn::eqz(max_indices), size, max_indices, output_memory_config);
+                    max_indices.deallocate();
+                    result = min(result, dim, output_memory_config);
```

Comment:
ttnn min or tteager ?

Reply:
ttnn composite min is not yet merged in main